### PR TITLE
Remove column and minor cleanup

### LINKED ISSUE #6740: Make classified listing categories configurable
This is currently hard-coded in the constant`ClassifiedListing::CATEGORIES_AVAILABLE`. In order for this to be useable by other communities, it needs to be configurable.

### PR DESCRIPTION
**Pull request by citizen428**
## What type of PR is this? (check all applicable)

- [X] Refactor

## Description

This PR removes the `category` column from the `classified_listings` table and includes some minor cleanups that were brought up during review of #7498.

## Related Tickets & Documents

Closes #6740. Tears of joy.

## Added tests?

- [X] no, because they aren't needed

## Added to documentation?

- [X] no documentation needed

## What gif best describes this PR or how it makes you feel?

owl

### CODE REVIEW

File: app/views/classified_listings/index.html.erb [hunk 1]
```diff
@@ -32,15 +32,15 @@
     data-allcategories="<%= categories_for_display.to_json %>"
     <% if @displayed_classified_listing %>
       data-displayedlisting="<%= @displayed_listing_json %> "
-    <% end %>
-  >
+    <% end %>>
```
Comment (citizen428) [inline app/views/classified_listings/index.html.erb:35]: It seems like one of our Husky hooks insists on reformatting this 🤷

File: db/schema.rb [hunk 2]
```diff
@@ -308,7 +308,6 @@
     t.string "slug"
     t.string "status", default: "active"
     t.datetime "updated_at", null: false
-    t.index ["slug"], name: "index_chat_channels_on_slug", unique: true
```
Comment (vaidehijoshi) [inline db/schema.rb:311]: Hmmm is this line unintentionally committed? 🤨

File: app/views/classified_listings/index.html.erb [hunk 3]
```diff
@@ -32,15 +32,15 @@
...
+
     <div class="classified-filters">
     <div class="classified-filters-categories">
-      <a href="/listings" class="<%= "selected" if params[:category].blank? %> data-no-instant="true">all</a>
+      <a href="<%= classified_listings_path %>" class="<%= "selected" if params[:category].blank? %> data-no-instant=" true ">all</a>
```
Comment (citizen428) [inline app/views/classified_listings/index.html.erb:39]: @rhymes Good idea using the path helpers, changed now.

File: app/views/classified_listings/index.html.erb [hunk 4]
```diff
@@ -32,15 +32,15 @@
...
       <% categories_for_display.each do |cat| %>
-        <a href="/listings/<%= cat[:slug] %>" class="<%= "selected" if params[:category] == cat[:slug] %> data-no-instant="true"><%= cat[:name] %></a>
+        <a href="<%= classified_listing_category_path(category: cat[:slug]) %>" class="<%= "selected" if params[:category] == cat[:slug] %> data-no-instant="true"><%= cat[:name] %></a>
```
Comment (citizen428) [inline app/views/classified_listings/index.html.erb:41]: Added this new one to `routes.rb`, see below.

File: db/schema.rb (hunk 2 above)
Comment (citizen428) [inline db/schema.rb:311]: @rhymes Time we remove this for good?

Comment (citizen428) [reply db/schema.rb:311]: Nope. This field should long been gone but keeps coming back.

Comment (vaidehijoshi) [reply db/schema.rb:311]: 👍👍 I’ve been seeing it show up intermittently but I thought it was just my local DB. Good riddance! :)

Comment (citizen428) [reply db/schema.rb:311]: @vaidehijoshi Here's a recent PR that also removed it and we should be careful to make sure it eventually stays gone: https://github.com/thepracticaldev/dev.to/pull/7528

File: app/views/classified_listings/index.html.erb (hunk 1 above)
Comment (rhymes) [reply app/views/classified_listings/index.html.erb:35]: let the tool do they :D

File: app/views/classified_listings/index.html.erb (hunk 3 above)
Comment (rhymes) [reply app/views/classified_listings/index.html.erb:39]: why the spaces around `true` ?

File: db/schema.rb (hunk 2 above)
Comment (rhymes) [reply db/schema.rb:311]: how did it come back again! yeah!

File: app/views/classified_listings/index.html.erb (hunk 3 above)
Comment (citizen428) [reply app/views/classified_listings/index.html.erb:39]: I don't now. Didn't even notice them, don't know where they came from. I'll remvoe them.

**Pull request merged**